### PR TITLE
Add integration test for GCSDoneFileMarker

### PR DESCRIPTION
**Pull request by dardanxh**
Jira ticket: https://cdap.atlassian.net/browse/PLUGIN-558

### CODE REVIEW

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java [hunk 1]
```diff
@@ -432,6 +434,41 @@ public void testGCSMoveRecursive() throws Exception {
...
+    ApplicationId appId = TEST_NAMESPACE.app("GCSDoneFileMarkerTest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    // start the pipeline and wait for it to finish
+    startWorkFlow(appManager, ProgramRunStatus.COMPLETED);
+
+    // bucket2 should have all content from bucket1 plus an empty DONE file
+    assertGCSContents(bucket2, "1.txt", "1");
+    assertGCSContents(bucket2, "2.txt", "2");
+    assertGCSContents(bucket2, "DONE", "");
```
Comment (CuriousVini) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: can this be `assertGCSContents(bucket2, "DONE")`?

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java [hunk 2]
```diff
@@ -432,6 +434,41 @@ public void testGCSMoveRecursive() throws Exception {
     assertGCSContents(bucket2, "2.txt", "2");
   }
 
+  @Test
+  public void testGCSDoneFileMarker() throws Exception {
```
Comment (CuriousVini) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: please also add a test for failure case

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java [hunk 3]
```diff
@@ -432,6 +434,41 @@ public void testGCSMoveRecursive() throws Exception {
...
+    String prefix = "cdap-gcs-done-file-marker";
+    String bucket1Name = String.format("%s-1-%s", prefix, UUID.randomUUID());
+    String bucket2Name = String.format("%s-2-%s", prefix, UUID.randomUUID());
+    String doneFileMarkerPath = bucket2Name + File.separator + "DONE";
+
+    Bucket bucket1 = createBucket(bucket1Name);
```
Comment (CuriousVini) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: are these buckets deleted after the test completes?

Comment (dardanxh) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: Yes, the bucket is marked for deletion and will get deleted when the test completes. This functionality is provided at `createBucket` method.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java (hunk 1 above)
Comment (dardanxh) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: Changed this to `assertExists(destinationBucket, MARKER_FILE_NAME)`.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java (hunk 3 above)
Comment (CuriousVini) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: thanks!

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java [hunk 4]
```diff
@@ -566,6 +645,16 @@ private ETLStage createMoveStage(String name, String src, String dest, boolean r
                                             GOOGLE_CLOUD_ARTIFACT));
   }
 
+  private ETLStage createGCSFileMarkerStage(String name, String path, String runCondition) {
+    return new ETLStage(name, new ETLPlugin(GCS_DONE_FILE_MARKER_PLUGIN_NAME, Action.PLUGIN_TYPE,
+                                            ImmutableMap.of("project", getProjectId(),
+                                                            "path", path,
+                                                            "runCondition", runCondition
+                                            ),
```
Comment (CuriousVini) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: nit: can be moved to previous line

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java [hunk 5]
```diff
@@ -566,6 +645,16 @@ private ETLStage createMoveStage(String name, String src, String dest, boolean r
...
+                                            GOOGLE_CLOUD_ARTIFACT
```
Comment (CuriousVini) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: nit: can be moved to previous line

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java [hunk 6]
```diff
@@ -566,6 +645,16 @@ private ETLStage createMoveStage(String name, String src, String dest, boolean r
...
+    ));
```
Comment (CuriousVini) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java]: nit: can be moved to previous line

**Review by CuriousVini — COMMENTED** [review]
Please make one of the tests to have all config properties as macro

**Review by CuriousVini — COMMENTED** [review]
Please use all the fields as macro in the success file creation test. In another test, include only bucket name as a macro

**Comment by dardanxh** [comment]
> Please use all the fields as macro in the success file creation test. In another test, include only bucket name as a macro

@CuriousVini could you share a reference example where the macros are used?

**Comment by CuriousVini** [comment]
> > Please use all the fields as macro in the success file creation test. In another test, include only bucket name as a macro
> 
> @CuriousVini could you share a reference example where the macros are used?

https://github.com/cdapio/cdap-integration-tests/pull/1084/files#diff-1336e750afbf76a0a305f1fc8bacd8ab942bca785bdc463d7844b4684131e688R345

**Comment by dardanxh** [comment]
> Please use all the fields as macro in the success file creation test. In another test, include only bucket name as a macro

@CuriousVini In the success file creation test I have three buckets:
1. Source bucket
2. Destination bucket
3. File marker bucket (which can be same as the destination bucket).


With *"In another test, include only bucket name as a macro"* you mean to consider as macro all of these buckets, or only the File marker bucket?

**Review by CuriousVini — APPROVED** [review]
please squash the commits

**Comment by dardanxh** [comment]
> please squash the commits

@CuriousVini I made the final changed. Please check.

**Review by CuriousVini — APPROVED** [review]

**Comment by CuriousVini** [comment]
noticing build errors:
```
[ERROR] /home/travis/build/cdapio/cdap-integration-tests/integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GCSTest.java:465:12: Name 'RUN_CONDITION' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
```

**Comment by CuriousVini** [comment]
can you fix them?

**Comment by dardanxh** [comment]
> can you fix them?

@CuriousVini 😬 Fixed.

**Review by CuriousVini — APPROVED** [review]

**Comment by CuriousVini** [comment]
@dardanxh please add the jira link in the description and add this PR to the jira

**Comment by dardanxh** [comment]
> @dardanxh please add the jira link in the description and add this PR to the jira

@CuriousVini added jira link in description. 

Waiting for the update of permissions in my jira account. Will comment the PR link there as soon as I have the permissions in place.

**Review by CuriousVini — APPROVED** [review]

**Pull request merged**